app-backend/tool/src/main/scala/eval/Interpreter.scala

```diff
@@ -73,6 +73,13 @@ object Interpreter extends LazyLogging {
     case o: Operation => o.args.map(a => overrideParams(a, overrides)).sequence.map(ks => o.withArgs(ks))
   }
 
+  /** Does a given AST have at least one source? */
+  private def hasSources[M: Monoid](ast: MapAlgebraAST): Interpreted[M] = {
```

Comment:
More generalized type checking in MAML (DSL type-checking, not Scala) will take care of this automatically, I'm thinking.